integration/spark/README.md

```diff
@@ -0,0 +1,147 @@
+# OpenLineage Spark Agent
+
+The OpenLineage Spark Agent uses jvm instrumentation to emit OpenLineage metadata.
+
+## Installation
+
+Maven:
+
+```xml
+<dependency>
+    <groupId>io.openlineage</groupId>
+    <artifactId>openlineage-spark</artifactId>
+    <version>0.1.0</version>
+</dependency>
+```
+
+or Gradle:
+
+```groovy
+implementation 'io.openlineage:openlineage-spark:0.1.0'
+```
+
+## Getting started
+
+### Quickstart
+The fastest way to get started testing Spark and OpenLineage is to use the docker-compose files included
+in the project. From the spark integration directory ($OPENLINEAGE_ROOT/integrations/spark) execute
+```bash
+docker-compose up
+```
+This will start a Jupyter Spark notebook on localhost:8888. On startup, the notebook container logs will show a list of URLs
+including an access token, such as
+```bash
+notebook_1  |     To access the notebook, open this file in a browser:
+notebook_1  |         file:///home/jovyan/.local/share/jupyter/runtime/nbserver-9-open.html
+notebook_1  |     Or copy and paste one of these URLs:
+notebook_1  |         http://abc12345d6e:8888/?token=XXXXXX
+notebook_1  |      or http://127.0.0.1:8888/?token=XXXXXX
+```
+Copy the URL with the localhost IP and paste into your browser window to begin creating a new Jupyter
+Spark notebook (see the [https://jupyter-docker-stacks.readthedocs.io/en/latest/](docs) for info on
+using the Jupyter docker image).
+
+#OpenLineageSparkListener as a plain Spark Listener
+The SparkListener can be referenced as a plain Spark Listener implementation.
+
+Create a new notebook and paste the following into the first cell:
+```python
+from pyspark.sql import SparkSession
+
+spark = (SparkSession.builder.master('local')
+         .appName('sample_spark')
+         .config('spark.jars.packages', 'io.github.openlineage:openlineage-spark-0.1-SNAPSHOT.jar'
+         .config('spark.extraListeners', 'openlineage.spark.agent.OpenLineageSparkListener')
+         .config('openlineage.url', 'http://marquez:5000/api/v1/namespaces/spark_integration/')
```

Comment:
The config `openlineage.url` should have the `spark.*` prefix

Reply:
I guess not, it's the last config which wasn't wrapped with "spark." prefix. 
[marquez](https://github.com/MarquezProject/marquez/blob/main/integrations/spark/src/main/java/marquez/spark/agent/SparkListener.java#L262)
I was thinking to change that, but maybe in next PR, don't want to mix things here

Reply:
so, it's not working with spark.openlineage.url.
```
 "--conf",
 "spark.openlineage.url=" + "http://openlineageclient:1080/api/v1",
```
I've changed SparkContainerIntegrationTest to use url, and it fails with 
` org.apache.hc.core5.http.ProtocolException: Target host is not specified)`
So, I guess we should fix that